packages/react-dom/src/client/ReactDOMFiberComponent.js

```diff
@@ -823,57 +724,26 @@ export function diffHydratedProperties(
 
   // TODO: Make sure that we check isMounted before firing any of these events.
   switch (tag) {
-    case 'iframe':
-    case 'object':
-      trapBubbledEvent('topLoad', 'load', domElement);
-      break;
-    case 'video':
-    case 'audio':
-      // Create listener for each media event
-      for (var event in mediaEvents) {
-        if (mediaEvents.hasOwnProperty(event)) {
-          trapBubbledEvent(event, mediaEvents[event], domElement);
-        }
-      }
-      break;
-    case 'source':
-      trapBubbledEvent('topError', 'error', domElement);
-      break;
-    case 'img':
-    case 'image':
-      trapBubbledEvent('topError', 'error', domElement);
-      trapBubbledEvent('topLoad', 'load', domElement);
-      break;
-    case 'form':
-      trapBubbledEvent('topReset', 'reset', domElement);
-      trapBubbledEvent('topSubmit', 'submit', domElement);
```

Comment:
for these, is there a reason why we were calling them out specifically here vs in the `ensureListeningTo` above? Do they need to probably be attached like `onChange` now?

Reply:
As far as I can tell, this is just a pre-emptive attachment because `ensureListeningTo` attaches on the owner document.

I think there is a change we could send to master here. We can attach `reset`, `submit`, `load`, etc... directly on the element. I do not think the savings is substantial because these events are almost only ever added to these elements (when was the last time you attached `onError` on a div or button?)

I'm going to check to see if we still need `onChange` to attach here.

Reply:
☝️ edited with some clarification.

Reply:
Made it consistent in 68b42c2

Reply:
ah so since everything is being attached directly there is no need to handle it here anymore 👍  right